Orange/widgets/data/owsave.py

```diff
@@ -126,18 +126,24 @@ def adjust_label(self):
 
     @Inputs.data
     def dataset(self, data):
-        self.data = data
         self.save.setDisabled(data is None)
         self.save_as.setDisabled(data is None)
-        if data is not None:
-            self.save_file()
+        if data is None:
+            return
 
-        self.controls.filetype.clear()
-        if self.data.is_sparse():
-            self.controls.filetype.insertItems(0, [item for item, _, supports_sparse in FILE_TYPES
-                                                   if supports_sparse])
+        if data.is_sparse():
+            items = [item for item, _, supports_sparse in FILE_TYPES
+                     if supports_sparse]
         else:
-            self.controls.filetype.insertItems(0, [item for item, _, _ in FILE_TYPES])
+            items = [item for item, _, _ in FILE_TYPES]
```

Comment:
OK, that's one way to do it. It may be a matter of opinion, but isn't

```
items = [item for item, _, supports_sparse in FILE_TYPES
         if supports_sparse or not data.is_sparse()]
```

more concise and shorter than if's with separate comprehensions? :)